CourseSearchBuilderBlobTrigger/search.py

```diff
@@ -0,0 +1,355 @@
+import logging
+import requests
+
+from SharedCode import exceptions
+from . import models
+
+def delete_index(url, headers, index_name):
+	try:
+		response = requests.delete(url, headers=headers)
+	except requests.exceptions.RequestException as e:
+		logging.exception('unexpected error deleting index',exc_info=True)
+		raise exceptions.StopEtlPipelineErrorException(e)
+		
+	if response.status_code == 204:
+		logging.warn(f'course search index already exists, successful deletion prior to recreation\n\
+			index: {index_name}')
+	elif response.status_code != 404:
+        # 404 is the expected response when deleting a course search index
```

Comment:
or perhaps "404 is the expected response, because normally the course search index will not exist"